coredhcp.go

```diff
@@ -101,27 +102,75 @@ func (s *Server) LoadPlugins(conf *config.Config) ([]*plugins.Plugin, []*plugins
 // It will not reply if the resulting response is `nil`.
 func (s *Server) MainHandler6(conn net.PacketConn, peer net.Addr, req dhcpv6.DHCPv6) {
 	var (
-		resp dhcpv6.DHCPv6
-		stop bool
+		resp, tmp dhcpv6.DHCPv6
+		stop      bool
+		err       error
 	)
+
+	// Create a suitable basic response packet
+	switch req.Type() {
+	case dhcpv6.MessageTypeSolicit:
+		tmp, err = dhcpv6.NewAdvertiseFromSolicit(req)
+	case dhcpv6.MessageTypeRequest, dhcpv6.MessageTypeConfirm, dhcpv6.MessageTypeRenew,
+		dhcpv6.MessageTypeRebind, dhcpv6.MessageTypeRelease, dhcpv6.MessageTypeInformationRequest:
+		tmp, err = dhcpv6.NewReplyFromDHCPv6Message(req)
+	default:
+		err = fmt.Errorf("MainHandler6: message type %d not supported", req.Type())
+	}
+
+	if err != nil {
+		log.Printf("MainHandler6: NewReplyFromDHCPv6Message failed: %v", err)
+		return
+	}
+	resp = tmp
 	for _, handler := range s.Handlers6 {
 		resp, stop = handler(req, resp)
 		if stop {
 			break
 		}
 	}
+
 	if resp != nil {
 		if _, err := conn.WriteTo(resp.ToBytes(), peer); err != nil {
-			log.Printf("conn.Write to %v failed: %v", peer, err)
+			log.Printf("MainHandler6: conn.Write to %v failed: %v", peer, err)
 		}
 	} else {
-		log.Print("Dropping request because response is nil")
+		log.Print("MainHandler6: dropping request because response is nil")
 	}
 }
 
 // MainHandler4 is like MainHandler6, but for DHCPv4 packets.
-func (s *Server) MainHandler4(conn net.PacketConn, peer net.Addr, d *dhcpv4.DHCPv4) {
-	log.Print(d.Summary())
+func (s *Server) MainHandler4(conn net.PacketConn, peer net.Addr, req *dhcpv4.DHCPv4) {
+	var (
+		resp, tmp *dhcpv4.DHCPv4
+		err       error
+		stop      bool
+	)
+	if req.OpCode != dhcpv4.OpcodeBootRequest {
+		log.Print("MainHandler4: unsupported opcode %d. Only BootRequest (%d) is supported", req.OpCode, dhcpv4.OpcodeBootRequest)
```

Comment:
Build in Go 1.11 fails with:
```
./coredhcp.go:150: Logger.Print call has possible formatting directive %d
```

Maybe we should print the string with req.OpCode.String().

Reply:
the actual error is that I used `Print` instead of `Printf`